packages/token-generation-readmodel-checker-verifier/src/utils/utils.ts

```diff
@@ -0,0 +1,1254 @@
+      readModel,
+    });
+  const clientDifferences = await compareReadModelClientsWithTokenGenReadModel({
+    platformStatesEntries: platformClientEntries,
+    tokenGenerationStatesEntries,
+    readModel,
+  });
+
+  return (
+    countPurposeDifferences(purposeDifferences, loggerInstance) +
+    countAgreementDifferences(agreementDifferences, loggerInstance) +
+    countCatalogDifferences(catalogDifferences, loggerInstance) +
+    countClientDifferences(clientDifferences, loggerInstance)
+  );
+}
+
+// purposes
+export async function compareReadModelPurposesWithTokenGenReadModel({
+  platformStatesEntries,
+  tokenGenerationStatesEntries,
+  readModel,
+}: {
+  platformStatesEntries: PlatformStatesPurposeEntry[];
+  tokenGenerationStatesEntries: TokenGenerationStatesClientPurposeEntry[];
+  readModel: ReadModelRepository;
+}): Promise<PurposeDifferencesResult> {
+  const readModelService = readModelServiceBuilder(readModel);
+  const [resultsA, resultsB, resultsC] = await Promise.all([
+    platformStatesEntries,
+    tokenGenerationStatesEntries.filter(
+      (e) => TokenGenerationStatesClientPurposeEntry.safeParse(e).success
+    ),
+    readModelService.getAllReadModelPurposes(),
+  ]);
+
+  return zipPurposeDataById(
+    resultsA,
+    resultsB,
+    resultsC
+  ).reduce<PurposeDifferencesResult>((acc, [a, b, c]) => {
+    if (!c) {
+      // eslint-disable-next-line functional/immutable-data
+      acc.push([
+        a ? PartialPlatformStatesPurposeEntry.parse(a) : undefined,
+        b && b.length > 0
+          ? PartialTokenStatesPurposeEntry.array().parse(b)
+          : undefined,
+        undefined,
+      ]);
+      return acc;
+    }
+
+    const purposeState = getPurposeStateFromPurposeVersions(c.versions);
+    const lastPurposeVersion = getLastPurposeVersion(c.versions);
+
+    const { status: isPlatformStatesCorrect, data: platformPurposeEntryDiff } =
+      validatePurposePlatformStates({
+        platformPurposeEntry: a,
+        purpose: c,
+        purposeState,
+        lastPurposeVersion,
+      });
+
+    const {
+      status: isTokenGenerationStatesCorrect,
+      data: tokenPurposeEntryDiff,
+    } = validatePurposeTokenGenerationStates({
+      tokenEntries: b,
+      purpose: c,
+      purposeState,
+      lastPurposeVersion,
+    });
+
+    if (!isPlatformStatesCorrect || !isTokenGenerationStatesCorrect) {
+      // eslint-disable-next-line functional/immutable-data
+      acc.push([
+        platformPurposeEntryDiff,
+        tokenPurposeEntryDiff,
+        PartialPurpose.parse(c),
+      ]);
+    }
+
+    return acc;
+  }, []);
+}
+
+function validatePurposePlatformStates({
+  platformPurposeEntry,
+  purpose,
+  purposeState,
+  lastPurposeVersion,
+}: {
+  platformPurposeEntry: PlatformStatesPurposeEntry | undefined;
+  purpose: Purpose;
+  purposeState: ItemState;
+  lastPurposeVersion: PurposeVersion;
+}): {
+  status: boolean;
```

Comment:
which is the meaning of `status`? may I ask to rename it to match the meaning of the boolean?